Training/Training/intlist.c

```diff
@@ -0,0 +1,221 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include "intlist.h"
+#include <stdbool.h>
+
+bool isDeleted; //flag variable for list deleted status
+
+/// <summary>Creates a new LinkedList to hold integers</summary>
+/// <returns>pointer to linked list</returns>
+LinkedList* Create () {
+	LinkedList* list = (LinkedList*)malloc (sizeof (LinkedList));
+	if (list != NULL) {
+		list->head = NULL;
+		list->isDeleted = false;
+	}
+	return list;
+}
+
+
+/// <summary>Adds element to the end of the list</summary>
+/// <param name="linkedlist pointer"></param>
+/// <param name="value"></param>
+/// <returns>Error code (if any)</returns>
+int Add (LinkedList* list, dataType data) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: checks if the list is empty
+	Node* newNode = CreateNode (data);
+	if (list->head == NULL) {
+		list->head = newNode;
+		return 0;
+	}
+	//case2: if the list is not empty, it iterates to the desired position
+	Node* current = list->head;
+	while (current->next != NULL) {
+		current = current->next;
+	}
+	current->next = newNode;
+	return 0;
+
+}
+
+
+/// <summary>Return the number of elements in list</summary>
+/// <param name="linkedlist pointer"></param>
+/// <returns>Error code (if any)</returns>
+int Count (LinkedList* list) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: if the list is not deleted, it iterates to get count
+	int count = 0;
+	Node* current = list->head;
+	while (current != NULL) {
+		count++;
+		current = current->next;
+	}
+	return count;
+}
+
+
+/// <summary>Delete the entire list</summary>
+/// <param name="linkedlist pointer"></param>
+/// <returns>Error code (if any)</returns>
+int Delete (LinkedList* list) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: Checks if the list is empty and deletes list
+	Node* current = list->head;
+	Node* next = NULL;
+	if (current == NULL) {
+		list->isDeleted = true;
+		return 0;
+	}
+	//case2: if the list is not empty, it iterates to the end and deletes all nodes
+	while (current != NULL) {
+		next = current->next;
+		free (current);
+		current = next;
+	}
+	list->head = NULL;
+	list->isDeleted = true;
+	return 0;
+}
+
+
+/// <summary>Removes the first occurrence of specified value in list</summary>
+/// <param name="linkedlist pointer"></param>
+/// <param name="value"></param>
+/// <returns>Error code (if any)</returns>
+int Remove (LinkedList* list, dataType data) {
+	//BaseCase: checks if the list is deleted
+	if (list->isDeleted == true) return E_LIST_DOES_NOT_EXIST;
+	//case1: Checks if the list is empty
+	Node* temp = list->head;
+	Node* prev = NULL;
+	if (temp == NULL) return E_EMPTY_LIST;
+	//case2: Check if the data in head(1st node) is to be removed
+	if (temp != NULL && temp->data == data) {

```

Comment:
You've already check for temp == NULL and are returning. You don't need to check again.
This if condition as also not required. Your while loop will handle it anyway.